### ISSUE TITLE
Uncaught (in promise) TypeError: Cannot set property 'error' of undefined

### ISSUE THREAD

**Issue by em-ka** [final-form/react-final-form#49]
Hi, thank you for your effort on this new promising library. I've made some tests if it's suitable for my project and found only this issue:

### Are you submitting a **bug report** or a **feature request**?

A bug report.

### What is the current behavior?

When I'm using asynchronous field-level validation everything works fine except on Form unmounting (the form is successfully submitted and the page is changed via react router) an error will rise:

```
final-form.es.js:561 Uncaught (in promise) TypeError: Cannot set property 'error' of undefined
    at final-form.es.js:561
    at Array.forEach (<anonymous>)
    at processErrors (final-form.es.js:557)
    at final-form.es.js:581
    at <anonymous>
```

When I look into the `final-form.es.js:561` and add a check if `fields[name]` is not `undefined` everything works fine.

Last form values before the error rises is according to the `debug={console.log}` an empty object `{}`.

### Sandbox Link

I was not able to reproduce it on any simplified example yet.

### What's your environment?

RFF 1.2.0, FF 1.3.0

**Comment by erikras** [comment]
Fix published in [`final-form v1.3.1`](https://github.com/final-form/final-form/releases/tag/v1.3.1).

**Comment by em-ka** [comment]
@erikras, you rock!
